bot/exts/fun/off_topic.py

```diff
@@ -184,56 +193,49 @@ async def list_ot_names(self, ctx: Context) -> None:
     async def update_ot_channel_name(self) -> None:
-        """Update ot-channel name everyday at midnight."""
-        while not self.bot.is_closed():
-            now = datetime.utcnow()
-            midnight_datetime = datetime.combine(
-                now.date() + timedelta(days=1), time(0)
-            )
-            logger.info(
-                f"Waiting until {midnight_datetime} for re-naming off-topic channel name."
-            )
-            await sleep_until(midnight_datetime)
-
-            # Algorithm to select next off topic name based on usage/number of times the name has been used.
-            # Least used names have a higher chance of being selected.
-            usage = set(self.ot_names.values())
-            distribution = [1 / (i + 1) for i in usage]
-            chosen_usage = random.choices(list(usage), distribution)[0]
-
-            chosen_ot_names = [
-                name for name, usage in self.ot_names.items() if usage == chosen_usage
-            ]
-
-            def to_channel_name(ot_name: str) -> str:
-                """Append off topic name prefix."""
-                return f"{OT_NAME_PREFIX}{ot_name}"
-
-            def from_channel_name(ot_name: str) -> str:
-                """Detach off topic name prefix."""
-                return ot_name[len(OT_NAME_PREFIX) :]
-
-            current_name = from_channel_name(self.ot_channel.name)
-            new_name = current_name
-
-            while new_name == current_name:
-                new_name = random.choice(chosen_ot_names)
-
-            self.ot_names[new_name] += 1
-            await db_execute(
-                self.bot.db_pool,
-                "UPDATE offtopicnames SET num_used=num_used+1 WHERE name=$1",
-                new_name,
-            )
+        """Update ot-channel name everyday at `REFRESH_TIMES`."""
+        # await the cache method to ensure we pull from an up-to-date cache every time
+        # this also fixes a race condition where the cache wasn't populated
+        await self._cache()
+        # Algorithm to select next off topic name based on usage/number of times the name has been used.
+        # Least used names have a higher chance of being selected.
+        usage = set(self.ot_names.values())
+        distribution = [1 / (i + 1) for i in usage]
+        chosen_usage = random.choices(list(usage), distribution)[0]
+
+        chosen_ot_names: set[str] = {
+            name for name, usage in self.ot_names.items() if usage == chosen_usage
+        }
+
+        def to_channel_name(ot_name: str) -> str:
+            """Append off topic name prefix."""
+            return f"{OT_NAME_PREFIX}{ot_name}"
+
+        def from_channel_name(ot_name: str) -> str:
+            """Detach off topic name prefix."""
+            return ot_name.removeprefix(OT_NAME_PREFIX)
+
+        current_name = from_channel_name(self.ot_channel.name)
+
+        chosen_ot_names.discard(current_name)
+        new_name = random.choice(list(chosen_ot_names))
```

Comment:
```suggestion
        new_name = random.choice(tuple(chosen_ot_names))
```
This might end up being a nitpick, but since you don't need to modify the list anywhere, why not just use a tuple? There